<?xml version="1.0"?>
<launch>

  <arg name="model_config" default=""/>
  <arg name="camera_topic" default="/kinect2/qhd/image_color_rect"/>
  <arg name="camera_info_topic" default="/kinect2/qhd/camera_info"/>
  <arg name="fixed_frame" default="odom"/>
  <arg name="tracking" default="true"/>
  <arg name="filter_boxes" default="true"/>
  <arg name="inside_box_ratio" default="0.8"/>
  <arg name="rviz" default="true"/>
  
  <!-- detection and tracking module --> 
  <node name="mobilityais_detector" type="detection_tracking.py" pkg="mobilityaids_detector" output="screen">
      <param name="model_config" type="string" value="$(arg model_config)"/>
      <param name="camera_topic" type="string" value="$(arg camera_topic)"/>
      <param name="camera_info_topic" type="string" value="$(arg camera_info_topic)"/>
      <param name="fixed_frame" type="string" value="$(arg fixed_frame)"/>
      <param name="tracking" type="bool" value="$(arg tracking)"/>
      <param name="filter_inside_boxes" type="bool" value="$(arg filter_boxes)"/>
      <param name="inside_box_ratio" type="double"  value="$(arg inside_box_ratio)"/>
  </node>

  <!-- start rviz visualization -->
  <group if="$(arg rviz)">
    <group if="$(arg tracking)">
      <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find mobilityaids_detector)/cfg/config_tracking.rviz"/>
    </group> 
    <group unless="$(arg tracking)">
      <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find mobilityaids_detector)/cfg/config_detection.rviz"/>
    </group>
  </group>

</launch>
